x/crosschain/keeper/msg_server_add_to_outtx_tracker_test.go

```diff
@@ -17,89 +22,173 @@ func getEthereumChainID() int64 {
 	return 5 // Goerli
 }
 
-// setEnabledChain sets the chain as enabled in chain params
-func setEnabledChain(ctx sdk.Context, zk keepertest.ZetaKeepers, chainID int64) {
-	zk.ObserverKeeper.SetChainParamsList(ctx, observertypes.ChainParamsList{ChainParams: []*observertypes.ChainParams{
-		{
-			ChainId:                  chainID,
-			ConnectorContractAddress: sample.EthAddress().Hex(),
-			BallotThreshold:          sdk.OneDec(),
-			MinObserverDelegation:    sdk.OneDec(),
-			IsSupported:              true,
-		},
-	}})
-}
+// TODO: Add a test case with proof and Bitcoin chain
+// https://github.com/zeta-chain/node/issues/1994
 
-// setupTssAndNonceToCctx sets tss and nonce to cctx
-func setupTssAndNonceToCctx(k *keeper.Keeper, ctx sdk.Context, chainId, nonce int64, status types.CctxStatus) {
-	tssPubKey := "zetapub1addwnpepq28c57cvcs0a2htsem5zxr6qnlvq9mzhmm76z3jncsnzz32rclangr2g35p"
-	k.GetObserverKeeper().SetTSS(ctx, observertypes.TSS{
-		TssPubkey: tssPubKey,
-	})
-	k.GetObserverKeeper().SetPendingNonces(ctx, observertypes.PendingNonces{
-		Tss:       tssPubKey,
-		NonceLow:  0,
-		NonceHigh: 1,
-		ChainId:   chainId,
-	})
-	cctx := types.CrossChainTx{
-		Creator: "any",
-		Index:   "0x123",
-		CctxStatus: &types.Status{
-			Status: status,
-		},
+func TestMsgServer_AddToOutTxTracker(t *testing.T) {
+
+	// mockCctxByNonce mocks the methods called by CctxByNonce to directly return the given cctx or error
+	mockCctxByNonce := func(
```

Comment:
Would be a good idea to move this function into the test-utils package

Reply:
a5f9653a